cocos/loop/SeqItem.h

```diff
@@ -0,0 +1,47 @@
+/****************************************************************************
+Copyright (c) 2017-2018 Xiamen Yaji Software Co., Ltd.
+
+http://www.cocos2d-x.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+namespace cocos2d
+{
+    namespace loop
+    {
+        template<typename T>
+        struct SeqItem {
+            SeqItem(uint64_t id, const std::string &name, T &data) : data(data), id(id), name(name) {}
+            SeqItem(uint64_t id, const std::string &name, T &&data) : data(data), id(id), name(name) {}
+            SeqItem(uint64_t id, T &data) : data(data), id(id) {}
+            SeqItem(uint64_t id, T &&data) : data(data), id(id) {}
+            SeqItem(const SeqItem &d) :data(d.data), id(d.id), name(d.name) {}
+            T data;
+            uint64_t id;
+            std::string name = "";
```

Comment:
id is not initialized, and i think string is not needed to be initialized, it will be empty if not intialized